go/pkg/cache/singleflightcache.go

```diff
@@ -34,6 +34,20 @@ func (s *SingleFlight) LoadOrStore(key interface{}, valFn func() (interface{}, e
 	return e.val, e.err
 }
 
+// Load is similar to a sync.Map.Load.
+func (s *SingleFlight) Load(key interface{}) (val interface{}, err error, loaded bool) {
```

Comment:
I feel returning err in the last element is consistent.

Reply:
I am worried about an invalid interpretation of the signature. Typically, the returned err overrides the preceding return values, so the callsite checks err first. However in this case, `loaded` has higher precedence. The client side must check `loaded` first, and only then `err`, and then `val.` I am worried that if we return `val, loaded, err,` then the programmer would check values in the wrong order. WDYT?

Reply:
I see.